### ISSUE TITLE
Parsing logic `ALTER SYSTEM SET` has some inconsistencies

### ISSUE THREAD

**Issue by aalexandrov** [MaterializeInc/materialize#16513]
### What version of Materialize are you using?

v0.35.0-dev (69d248592)

### How did you install Materialize?

Built from source

### What is the issue?

Both variants of the `ALTER SYSTEM SET` statement
```sql
ALTER SYSTEM SET allowed_cluster_replica_sizes = '3xsmall';
ALTER SYSTEM SET allowed_cluster_replica_sizes = "3xsmall";
ALTER SYSTEM SET allowed_cluster_replica_sizes = '3xsmall', '2xsmall';
```
produce the following error
```
parameter "allowed_cluster_replica_sizes" requires a "string list" value
```

However this works:

```sql
ALTER SYSTEM SET allowed_cluster_replica_sizes = "3xsmall", "2xsmall";
```

**Comment by ggnall** [comment]
@aalexandrov do you need surfaces to pick this up? I see it's still assigned to you but on our board. cc @chaas

**Comment by aalexandrov** [comment]
I was planing to pick this up as a skunkworks project this or next week. 

If you feel that it needs to be prioritized higher please feel free to assign this to somebody else.

**Comment by aalexandrov** [comment]
Some notes for whoever picks this up:

At the moment `Value::parse` can receive input that can be:
1. Generated ad-hoc by `sequence_alter_system_set` / `sequence_set_variable` by match statements on `SetVariableValue` of the enclosing plan.
2. Generated by `Value::format` on a `SystemVar` that is written to stash. This path is also used to format the output when handling `SHOW` statements.

We somehow need to ensure that the string representation produced by (1) and (2) aligns. If necessary, this might entail forking splitting off the two responsibilities of `Value::format` into two different `Value` methods.

**Comment by mjibson** [comment]
Closed by #17077 17658
